### ISSUE TITLE
[SCM-991] GitDiffConsumer cannot parse moved files

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1218]
**[Jean-Noel Rouvignac](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jnrouvignac)** opened **[SCM-991](https://issues.apache.org/jira/browse/SCM-991?redirect=false)** and commented

`GitDiffConsumer` keep outputting `Unparseable line:` when diffing a commit where a file was moved using `{}git mv{`}.

The `Unparseable line:` message is output by `[GitDiffConsumer.consumeLine()](https://github.com/apache/maven-scm/blob/master/maven-scm-providers/maven-scm-providers-git/maven-scm-provider-git-commons/src/main/java/org/apache/maven/scm/provider/git/command/diff/GitDiffConsumer.java)` because the code does not know how to handle the line starting with "similarity":

```java
diff --git a/org/pager/PagerTest.java b/org/pager/LocalPagerTest.java
similarity index 87%
rename from org/pager/PagerTest.java
rename to org/pager/LocalPagerTest.java
index 5d0f7b7705..90c87686c7 100644
--- a/org/pager/PagerTest.java
+++ b/org/pager/LocalPagerTest.java
@@ -30,7 +30,7 @@ import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 @Test
-public class PagerTest {
+public class LocalPagerTest {

```

I am using git version 2.25.1 .

---

**Attachments:**
- [SCM-991.patch](https://issues.apache.org/jira/secure/attachment/13045563/SCM-991.patch) (_5.75 kB_)

**Remote Links:**
- [GitHub Pull Request #151
](https://github.com/apache/maven-scm/pull/151)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Merci, will check next week.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [07d51d34ec386d527ecc584e67514ff862caca1e](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=07d51d34ec386d527ecc584e67514ff862caca1e).

**Comment by jira-importer** [comment]
**[Jean-Noel Rouvignac](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jnrouvignac)** commented

Thank you Michael!
